src/GafferML/Tensor.cpp

```diff
@@ -176,6 +179,14 @@ DataPtr dataFromValue( const Ort::Value &value )
 
 					result = d;
 				}
+				else if constexpr( std::is_same_v<ElementType, Ort::BFloat16_t> )
+				{
+					FloatVectorDataPtr d = new FloatVectorData();
+					std::vector<float> &v = d->writable();
+					v.reserve( count );
+					std::transform( data, data + count, v.begin(), []( const Ort::BFloat16_t &e ) { return float( e ); } );
```

Comment:
I think `v.size()` is still going to be zero after this. You've set the capacity with `reserve()`, but `std::transform()` isn't doing a push back so `size()` won't change. Maybe a `back_insert_iterator()` would do the job? Would be good to add some coverage for this in the unit tests.